File: 88-merge-sorted-array/merge-sorted-array.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        int insP = m+n-1;
        int p1 = m-1;
        int p2 = n-1;

        while (p1>=0 && p2>=0){
            if(nums1[p1] > nums2[p2]) {
            nums1[insP] = nums1[p1];
            p1--;
            insP--;
            } else {
            nums1[insP] = nums2[p2];
            p2--;
            insP--;
            }
        }
        while (p2 >=0){
            nums1[insP] = nums2[p2];
            p2--;
            insP--;
        }
         
    }
}

File: 88-merge-sorted-array/merge-sorted-array.py
class Solution(object):
   def merge(self, nums1, m, nums2, n):
    # Start filling from the end
    i = m - 1  # Pointer for nums1's valid part
    j = n - 1  # Pointer for nums2
    k = m + n - 1  # Pointer for end of nums1 (where to insert)

    # While there are elements in both arrays
    while i >= 0 and j >= 0:
        if nums1[i] > nums2[j]:
            nums1[k] = nums1[i]
            i -= 1
        else:
            nums1[k] = nums2[j]
            j -= 1
        k -= 1

    # If nums2 still has elements, put them in nums1
    while j >= 0:
        nums1[k] = nums2[j]
        j -= 1
        k -= 1
